src/java.base/share/classes/jdk/internal/foreign/abi/UpcallLinker.java

```diff
@@ -61,6 +61,9 @@ public class UpcallLinker {
         }
     }
 
+    private UpcallLinker() {
```

Comment:
This doesn't buy much since it's internal API.

Reply:
I think it is good practice anyhow. This signals that it is an uninstantiable utility class.

Reply:
I understand your point of view. However, for these changes we have agreed to commit to a policy of sticking to less controversial changes, to move things forward. Adding `final` and private constructors in impl classes is not widespread practice in JDK code, so we should hold off at the moment (and maybe revisit later).